pkg/ffcapi/transaction_receipt.go

```diff
@@ -21,15 +21,25 @@ import (
 )
 
 type TransactionReceiptRequest struct {
-	TransactionHash string `json:"transactionHash"`
+	TransactionHash string            `json:"transactionHash"`
+	IncludeLogs     bool              `json:"includeLogs"`
+	EventFilters    []fftypes.JSONAny `json:"eventFilters"`
+	Methods         []fftypes.JSONAny `json:"methods"`
+	ExtractSigner   bool              `json:"extractSigner"`
 }
 
-type TransactionReceiptResponse struct {
+type TransactionReceiptResponseBase struct {
```

Comment:
As I worked this through in combination with https://github.com/hyperledger/firefly-evmconnect/pull/141, I found I had to make a tweak to the Receipt structure on the FFCAPI interface.

We have a (now slightly ugly looking sadly) JSON merging structure on the `EventWithContext` object, that defines how serialization happens into the JSON merging static and custom bits of the object together.

However, this is on `apitype` _not_ `ffcapi` where we just have `Event` - which is one part of what's merged.

Without doing really complex refactoring, I found a compromise where I map in the FFTM layer between the types. The easiest way to do this was adding this here.